Change default of git.log.showGraph to 'always'

stefanhaller opened the pull request:
- **PR Description**

Change default of git.log.showGraph to 'always'; most people seem to prefer it to be on.

Fixes #3312.

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go generate ./...`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [ ] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

stefanhaller commented:
This breaks a lot of tests because I insisted on matching the whole line. 😄 This branch has two different ways of fixing them, I can't quite make up my mind which one I prefer.

jesseduffield commented on pkg/integration/tests/bisect/basic.go:
```diff
@@ -14,7 +14,7 @@ var Basic = NewIntegrationTest(NewIntegrationTestArgs{
 			CreateNCommits(10)
 	},
-	SetupConfig: func(cfg *config.AppConfig) {},
+	SetupConfig: func(cfg *config.AppConfig) { cfg.UserConfig.Git.Log.ShowGraph = "never" },
```
Could we have this on its own line? Just so that the diff is cleaner when another line is eventually added (same with the other instances)

stefanhaller replied:
Sure, changed in 7ae1b3b0686b.

jesseduffield reviewed (commented):
Code looks good though I'm not seeing any of the test fixes you mentioned

stefanhaller commented:
> Code looks good though I'm not seeing any of the test fixes you mentioned

Because you always look at the "All file changes" view. 😄 I fixed the tests in one way in the first commit, and then reverted those hunks and fixed them another way in the second commit. We could either drop the second commit, or squash it into the first. I really can't decide which way of fixing the tests I like better.

jesseduffield commented:
> Because you always look at the "All file changes" view. 😄

Classic me, haha. I understand the dilemma: right now it's a choice between changing the config to be non-default, meaning it's not representative of the typical user, and updating the test assertion to include the graph, which is irrelevant to what we're testing. But I think we should go for a third option: simply remove any reference to the commit author in the assertions. It's always going to be 'CI' anyway. The only reason we need to include the graph in the assertion at the moment is because it sits between the author and the commit subject. If we at some point write a test that really does want to assert on both the author and the subject we can use `.Contains("CI").Contains("commit subject")`

stefanhaller commented:
> If we at some point write a test that really does want to assert on both the author and the subject we can use `.Contains("CI").Contains("commit subject")`

That's missing the point: we include the author in these tests not because we want to assert that the author is correct, but because we want to assert that there's nothing in between the author and the subject.

We'd have to write this instead:
```
    DoesNotContain("*").Contains("commit 07"),
    DoesNotContain("*").Contains("commit 06"),
    Contains("*").Contains("commit 05"),
    DoesNotContain("*").Contains("commit 04").IsSelected(),
    Contains("*").Contains("commit 03").IsSelected(),
    DoesNotContain("*").Contains("commit 02").IsSelected(),
    DoesNotContain("*").Contains("commit 01"),
```
I find this terrible, I greatly prefer what we have now:
```
    Contains("CI commit 07"),
    Contains("CI commit 06"),
    Contains("CI * commit 05"),
    Contains("CI commit 04").IsSelected(),
    Contains("CI * commit 03").IsSelected(),
    Contains("CI commit 02").IsSelected(),
    Contains("CI commit 01"),
```

And thinking about this more, I think it's ok to turn off the graph for these tests, so I'm in favor of squashing the second commit into the first.

jesseduffield commented:
Ah I see. I'm happy with that
